lib/Doctrine/ORM/Tools/SchemaValidator.php

```diff
@@ -304,4 +349,66 @@ public function getUpdateSchemaList(): array
 
         return $schemaTool->getUpdateSchemaSql($allMetadata, true);
     }
+
+    /** @return list<string> containing the found issues */
+    private function validatePropertiesTypes(ClassMetadataInfo $class): array
```

Comment:
We have no choice but to use `ClassMetadataInfo` here for now because of the calling code, and that should be fine, although I considered using `if (PHP_VERSION_ID >= 70400 && $class instanceof ClassMetadata) {`… that would be a bit weird.

Reply:
I'm not sure I understand this comment. Is this because of a possible different Persistence version and `validateClass(ClassMetadataInfo $class)`?

Reply:
I'm saying that in the unique call site of this method, we have no guarantee that we have a CM and not just a CMI. This explains why we use this type declaration here.